nemo/collections/nlp/models/language_modeling/megatron/griffin/griffin_block.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from megatron.core.models.common.language_module.language_module import LanguageModule
+from megatron.core.transformer.custom_layers.transformer_engine import TENorm
+from megatron.core.transformer.spec_utils import build_module
+from megatron.core.transformer.transformer_config import TransformerConfig
+from torch import nn
+
+from nemo.collections.nlp.models.language_modeling.megatron.griffin.griffin_layer_spec import (
+    griffin_mqa_layer_with_transformer_engine_spec,
+    griffin_recurrent_layer_with_transformer_engine_spec,
+)
+
+
+def get_griffin_layers(num_layers):
+    dict_spec = {
+        "Recurrent_Layer": griffin_recurrent_layer_with_transformer_engine_spec,
+        "Attention_Layer": griffin_mqa_layer_with_transformer_engine_spec,
+    }
+
+    griffin_layers = []
+    for i in range(num_layers):
+        if i % 3 == 2:
+            griffin_layers.append(dict_spec["Attention_Layer"])
+        else:
+            griffin_layers.append(dict_spec["Recurrent_Layer"])
+
+    return griffin_layers
+
+
+def create_block(
+    config, layer_spec, layer_idx=None,
+):
+    block = build_module(layer_spec, config,)
+    block.layer_number = layer_idx + 1
```

Comment:
but layer idx default is None, +1 might error out?

Reply:
maybe do not default anything

Reply:
Resolved